test/net/sourceforge/kolmafia/request/GenericRequestTest.java

```diff
@@ -190,4 +195,38 @@ public void willLocallySuppressUpdate() {
       }
     }
   }
+
+  @Test
+  public void testTracksTowelAcquired() {
+    var cleanups =
+        new Cleanups(
+            withProperty("lastTowelAscension", -1),
+            withNextResponse(
+                new FakeHttpResponse<>(
+                    302, Map.of("location", List.of("choice.php?forceoption=0")), ""),
+                new FakeHttpResponse<>(
+                    200, html("request/test_request_haunted_bathroom_off_the_rack.html")),
+                new FakeHttpResponse<>(
+                    200,
+                    ""), // An api request is triggered by the above containing a charpane request.
+                // We swallow it.
+                new FakeHttpResponse<>(
+                    200, html("request/test_request_haunted_bathroom_towel.html"))));
+
+    try (cleanups) {
+      // Prove that the towel is not set to the current ascension before we start the test.
+      assertThat("lastTowelAscension", not(isSetTo(KoLCharacter.getAscensions())));
```

Comment:
This is unnecessary, you are just testing withProperty. You should use withAscensions(x)

Reply:
Oops I think you accidentally marked this as resolved